test/plot_methods/bar_plot_test.rb

```diff
@@ -0,0 +1,74 @@
+require_relative "../test_helper"
+
+class PlotMethodsBarPlotTest < Test::Unit::TestCase
+  sub_test_case("function-call style") do
+    test("given x and y") do
+      x = [1, 2, 3, 4, 5]
+      y = [10, 20, 30, 25, 15]
+      fig = Charty.bar_plot(x, y)
+      assert_instance_of(Charty::Plotters::BarPlotter, fig)
+      assert_equal(x, fig.x)
+      assert_equal(y, fig.y)
+      assert_equal(nil, fig.color)
+      assert_equal(nil, fig.data)
+      assert_equal(nil, fig.palette)
+      assert_equal(x, fig.group_names)
+      assert_equal(y.map{|v| [v] }, fig.plot_data)
```

Comment:
One `assert_equal` is better for easy to debug:

```ruby
assert_equal({
               x: x,
               y: y,
               color: nil,
               data: nil,
               palette: nil,
               group_names: x,
               plot_data: y.map{|v| [v] },
             },
             {
               x: fix.x,
               y: fix.y,
               color: fig.color,
               data: fig.data,
               palette: fig.palette,
               group_names: fig.group_names,
               plot_data: fig.plot_data,
             })
```

Reply:
Sounds reasonable.